### PR TITLE
Add config option for imgui font

### PR DESCRIPTION
**Pull request by Hibyehello**
Font size only applies to custom fonts not Default

### CODE REVIEW

**Review by ximk — APPROVED** [review]
Approving from a testing and compiling standpoint. **Do not merge until a thorough code review is done.**

File: Source/Core/VideoCommon/OnScreenUI.cpp [hunk 1]
```diff
@@ -73,6 +75,14 @@ bool OnScreenUI::Initialize(u32 width, u32 height, float scale)
   // Font texture(s).
   {
     ImGuiIO& io = ImGui::GetIO();
+    if(Config::Get(Config::MAIN_IMGUI_FONT) != "Default"){
```
Comment (malleoz) [inline Source/Core/VideoCommon/OnScreenUI.cpp:78]: Your indentation is messed up here.

**Review by malleoz — CHANGES_REQUESTED** [review]

File: Source/Core/Core/API/Gui.h [hunk 2]
```diff
@@ -16,6 +16,7 @@ using Vec2f = ImVec2;
 namespace API
 {
 
+static ImFont* font;
```
Comment (vabold) [inline Source/Core/Core/API/Gui.h:19]: `g_font`.

**Review by vabold — CHANGES_REQUESTED** [review]

File: Source/Core/Core/API/Gui.cpp [hunk 3]
```diff
@@ -96,7 +97,21 @@ void Gui::DrawCircleFilled(const Vec2f center, float radius, u32 color, int num_
 
 void Gui::DrawText(const Vec2f pos, u32 color, std::string text)
 {
-  GUI_DRAW_DEFERRED(AddText(pos, ARGBToABGR(color), text.c_str()));
+  int enable_outline = Config::Get(Config::MAIN_IMGUI_OUTLINE_RES);
+  if(enable_outline == 2 || enable_outline == 3) {
+	GUI_DRAW_DEFERRED(AddText(API::font, Config::Get(Config::MAIN_IMGUI_FONT_SIZE), ImVec2(pos.x+1, pos.y+1), ARGBToABGR(color^0x00FFFFFF), text.c_str()));
```
[the remark this answers is not in the document]
Comment (Hibyehello) [reply Source/Core/Core/API/Gui.cpp:102]: I'm not sure what'd be the best way

**Review by ximk — COMMENTED** [review]
I just have optional things to consider including/changing, nothing major.

File: Source/Core/Core/API/Gui.cpp [hunk 4]
```diff
@@ -96,7 +97,21 @@ void Gui::DrawCircleFilled(const Vec2f center, float radius, u32 color, int num_
 
 void Gui::DrawText(const Vec2f pos, u32 color, std::string text)
 {
-  GUI_DRAW_DEFERRED(AddText(pos, ARGBToABGR(color), text.c_str()));
+  Config::OutlineRes enable_outline = Config::Get(Config::MAIN_IMGUI_OUTLINE_RES);
+  if(enable_outline == Config::OutlineRes::Thick || enable_outline == Config::OutlineRes::Full) {
+	GUI_DRAW_DEFERRED(AddText(API::g_font, Config::Get(Config::MAIN_IMGUI_FONT_SIZE), ImVec2(pos.x+1, pos.y+1), ARGBToABGR(color^0x00FFFFFF), text.c_str()));
```
Comment (malleoz) [inline Source/Core/Core/API/Gui.cpp:102]: Your indentation is messed up here.

**Review by malleoz — CHANGES_REQUESTED** [review]

**Comment by Hibyehello** [comment]
@malleoz I think I've addressed everything you wanted addressed

**Review by malleoz — APPROVED** [review]

**Pull request merged**